src/main/java/maksym/fedorenko/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -86,12 +86,10 @@ public List<BookDtoWithoutCategories> findBooksByCategoryId(Long id, Pageable pa
                 .toList();
     }
 
-    private void addBookCategoriesIfPresent(List<Long> categoryIds, Book book) {
-        if (categoryIds != null) {
-            Set<Category> categories = new HashSet<>(
-                    categoryRepository.findAllById(categoryIds));
-            book.setCategories(categories);
-        }
+    private void addBookCategories(List<Long> categoryIds, Book book) {
+        Set<Category> categories = new HashSet<>(
+                categoryRepository.findAllById(categoryIds));
```

Comment:
you may leverage `getReference` method in order to create mapping between book and its categories

Reply:
I don't sure how I can use it, I try use it with stream but receive LazyException